### PR TITLE
Write-back to memcache in the background, don't block the query.

### PR DESCRIPTION
**Pull request by tomwilkie**
Part of #209 

Also, when I flushed the cached due to previous bugs, we started trying to write 2.5k chunks to the cache in parallel.  This change limits the parallelism with which we write to the cache.

### CODE REVIEW

File: chunk/chunk_cache.go [hunk 1]
```diff
@@ -30,6 +31,12 @@ var (
 		Help:      "Total count of number of corrupt chunks found in memcache.",
 	})
 
+	memcacheDroppedWriteBack = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "cortex",
+		Name:      "memcache_dropped_write_back",
+		Help:      "Total count of number of dropped write backs to memcache.",
```
Comment (juliusv) [inline chunk/chunk_cache.go]: "count of number" sounds a bit redundant :D

Comment (tomwilkie) [inline chunk/chunk_cache.go]: Removed.

File: chunk/chunk_cache.go [hunk 2]
```diff
@@ -156,20 +188,31 @@ func (c *Cache) StoreChunk(ctx context.Context, key string, buf []byte) error {
...
+
+func (c *Cache) writeBackLoop() {
+	defer c.wg.Done()
+
+	for {
+		select {
+		case bgWrite := <-c.bgWrites:
+			err := c.StoreChunk(context.Background(), bgWrite.key, bgWrite.buf)
+			if err != nil {
+				log.Errorf("Error write to memcache: %v", err)
```
Comment (juliusv) [inline chunk/chunk_cache.go]: write -> writing

**Comment by juliusv** [comment]
:+1:

**Pull request merged**